### ISSUE TITLE
Support fields that use fallback synthetic source in ESQL

### ISSUE THREAD

**Issue by lkts** [elastic/elasticsearch#108883]
### Description

As discovered in #108881 `geo_shape` field is supported in ESQL but uses a block loader that loads data from `_source`. #108881 adds synthetic source support for `geo_shape` using infrastructure from #107567 but does not include corresponding changes to block loader or any other changes needed for ESQL.

**Comment by lkts** [comment]
@nik9000 yes, i think this is mostly a duplicate of those, i can close it

**Comment by martijnvg** [comment]
This fixed via #115114
